services/visual_recognition/v3.js

```diff
@@ -62,301 +37,331 @@ module.exports = function (RED) {
+      node.error(message, msg);
+      return false;
+    }
+    node.service = watson.visual_recognition({
+      api_key: node.apikey,
       version: 'v3',
       version_date: '2016-05-19'
     });
-
-    switch(feature)
-    {
-      case 'classifyImage' : 
-        visualRecognition.classify(params, cbdone);
-        break;
-      case 'detectFaces':
-        visualRecognition.detectFaces(params, cbdone);
-        break;
-      case 'recognizeText':
-        visualRecognition.recognizeText(params, cbdone);
-        break;
-      case 'createClassifier':
-        visualRecognition.createClassifier(params, cbdone);
-        break;
-      case 'retrieveClassifiersList':
-        visualRecognition.listClassifiers(params, cbdone);
-        break;
-      case 'retrieveClassifierDetails':
-        visualRecognition.getClassifier(params, cbdone);
-        break;
-      case 'deleteClassifier':
-        visualRecognition.deleteClassifier(params, cbdone);
-        break;
-    }
+    return true;
   }
 
-  // This is the Watson Visual Recognition V3 Node
-  function WatsonVisualRecognitionV3Node (config) {
-    RED.nodes.createNode(this, config);
-    var node = this;
-    var message;
-    this.on('input', function (msg) 
-    {
-      // Check which single feature has been requested.
-      var feature = config['image-feature'];
-      node.status({});
-      // so there is at most 1 temp file at a time (did not found a better solution...)
-      temp.cleanup(); 
 
-      if (!msg.payload) {
-        this.status({fill:'red', shape:'ring', text:'missing payload'}); 
-        message = 'Missing property: msg.payload';
-        node.error(message, msg);
-        return;
+  function processResponse(err, body, feature, node, msg) {
+    if (err != null && body==null) {
+      node.status({fill:'red', shape:'ring', 
+        text:'call to watson visual recognition v3 service failed'}); 
+      msg.result = {};
+      if (err.code==null) {
+        msg.result['error']=err;
+      } else {
+        msg.result['error_code']= err.code;
+        if (!err.error) {
+          msg.result['error']= err.error;
+        }        
       }
-      if (feature !== 'retrieveClassifiersList' && feature !== 'retrieveClassifierDetails' 
-          && feature !== 'deleteClassifier')
-      {
-        if (typeof msg.payload === 'boolean' || typeof msg.payload === 'number') 
-        {
-          this.status({fill:'red', shape:'ring', text:'bad format payload'}); 
-          message = 'Bad format : msg.payload must be a URL string or a Node.js Buffer';
-          node.error(message, msg);
-          return;
-        }
+      node.error(err);
+      return;
+    } else if (err == null && body != null && body.images != null && 
+      body.images[0].error) {
+      node.status({fill:'red', shape:'ring', 
+                   text:'call to watson visual recognition v3 service failed'}); 
+      msg.result = {};
+      msg.result['error_id']= body.images[0].error.error_id;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'error_id' is not in camel case.](https://www.codacy.com/app/BetaWorks-NodeRED-Watson/node-red-node-watson/file/2737047177/issues/source?bid=3330694&fileBranchId=3352492#l121)